runtime/lua/vim/treesitter.lua

```diff
@@ -110,18 +112,12 @@ function M.get_parser(bufnr, lang, opts)
     bufnr = a.nvim_get_current_buf()
   end
 
-  if lang == nil then
-    local ft = vim.bo[bufnr].filetype
-    if ft ~= '' then
-      lang = M.language.get_lang(ft) or ft
-      -- TODO(lewis6991): we should error here and not default to ft
-      -- if not lang then
-      --   error(string.format('filetype %s of buffer %d is not associated with any lang', ft, bufnr))
-      -- end
-    else
-      if parsers[bufnr] then
-        return parsers[bufnr]
-      end
+  if !valid_lang(lang) then
```

Comment:
I don't think that's valid Lua 😅 
```suggestion
  if not valid_lang(lang) then
```